Support copy/paste in Electron

bryfox opened the pull request:
Mostly just pulling in the related change from Server here.

Including the built-in copy/paste menu items automatically adds keyboard shortcuts. This enables copy/paste across the entire app, which I assume is OK. The mac-specific handling is for the app menu item, as in [the electron docs](https://electronjs.org/docs/api/menu#main-process).

(Note that this also overrides Electron's default menu during development; if we were purposefully using different menus for dev/prod, then this may not be what we want.)

Fixes #489.

jthrilly commented:
Overriding the menu is fine, as we will need to start adding our own menu items soon anyway (potentially as part of a proper fix for #490 and when implementing #245).

I haven't checked this out, but can you tell me if "allowing copying and pasting anywhere" includes arbitrary text (paragraphs, headings, etc.)? One thing we specifically disallowed through CSS was arbitrary selection, due to UX concerns. As long as this respects that, I think you can merge it.

bryfox commented:
Yes, this respects `user-select: none`. I hadn't noticed that was in there. (Should we also set `cursor: default` so it doesn't look selectable?)

bryfox commented:
I just pushed another commit that does add reload & dev tools in the menu. Dev tools is already available from the sidebar menu, but this restores default behavior during development.

jthrilly commented:
Perfect.



> (Should we also set cursor: default so it doesn't look selectable?)

Yes, definitely. This would be a good improvement.

bryfox commented:
Sorry for piling on to this, but it's all kind of related:

- updated to `cursor: default` as noted above
- removed client setup of devtools shortcuts, since the shortcuts are now automatically provided by the native menu item

rebeccamadsen commented:
I thought we don't show the electron menu for the production version, which is why we had a devtools shortcut in our menu? Or did I miss something? I haven't tried this branch out yet, so may just not understand exactly.

bryfox commented:
Sorry, I think I was unclear in the description. This *will* now show the same menus in both dev & release builds. If that's not what we want I'll close this PR. ('Automatic' keyboard shortcuts won't work without the native menu items.) 

We could likely implement keyboard-shortcut-only paste with the [clipboard API](https://electronjs.org/docs/api/clipboard) and :focus selectors.

jthrilly commented:
I think for the time being, this is fine.
